x/hashlock/context_test.go

```diff
@@ -18,40 +17,45 @@ func TestContext(t *testing.T) {
 	random := weave.NewAddress(foo)
 
 	bg := context.Background()
-	cases := []struct {
+	cases := map[string]struct {
 		ctx   weave.Context
 		perms []weave.Condition
 		match []weave.Address
 		not   []weave.Address
 	}{
-		{bg, nil, nil, []weave.Address{sig, other, random}},
-		{
-			withPreimage(bg, foo),
-			[]weave.Condition{PreimageCondition(foo)},
-			[]weave.Address{other},
-			[]weave.Address{sig, random},
+		"empty context": {
+			ctx: bg,
+			not: []weave.Address{sig, other, random},
 		},
-		{
-			withPreimage(bg, []byte("one more time")),
-			[]weave.Condition{PreimageCondition([]byte("one more time"))},
-			nil,
-			[]weave.Address{sig, other, random},
+		"context with a preimage": {
+			ctx:   withPreimage(bg, foo),
+			perms: []weave.Condition{PreimageCondition(foo)},
+			match: []weave.Address{other},
+			not:   []weave.Address{sig, random},
+		},
+		"context with a preimage 2": {
+			ctx:   withPreimage(bg, []byte("one more time")),
+			perms: []weave.Condition{PreimageCondition([]byte("one more time"))},
+			not:   []weave.Address{sig, other, random},
 		},
 	}
 
 	auth := Authenticate{}
 
-	for i, tc := range cases {
-		t.Run(fmt.Sprintf("case-%d", i), func(t *testing.T) {
-			perms := auth.GetConditions(tc.ctx)
-			assert.Equal(t, tc.perms, perms)
+	for testName, tc := range cases {
+		t.Run(testName, func(t *testing.T) {
+			assert.Equal(t, tc.perms, auth.GetConditions(tc.ctx))
```

Comment:
Right now this will be Fatal, but the name suggests otherwise (and different from old behavior).

Reply:
This is a change also below but this condition is good enough to abort the test already.